patches/cygwin/0004-fix-gcc-15.patch

```diff
@@ -0,0 +1,2125 @@
+diff --git a/newlib/libc/posix/collate.c b/newlib/libc/posix/collate.c
+index a2f62c978..8e84895ba 100644
+--- a/newlib/libc/posix/collate.c
++++ b/newlib/libc/posix/collate.c
+@@ -58,8 +58,9 @@ struct __collate_st_chain_pri __collate_chain_pri_table[TABLE_SIZE];
+ void __collate_err(int ex, const char *f);
+ 
+ int
+-__collate_load_tables(encoding)
+-	char *encoding;
++__collate_load_tables(
```

Comment:
I'll discuss with Cygwin maintainers whether they would like to apply those changes or set `-std=gnu17` in the build scripts explicitly.

Reply:
Discussed the patch was sent to newlib and Cygwin mailing lists.